airflow/providers/amazon/aws/sensors/s3.py

```diff
@@ -144,32 +145,39 @@ def execute(self, context: Context) -> None:
             super().execute(context)
         else:
             if not self.poke(context=context):
-                self.defer(
-                    timeout=timedelta(seconds=self.timeout),
-                    trigger=S3KeyTrigger(
-                        bucket_name=cast(str, self.bucket_name),
-                        bucket_key=self.bucket_key,
-                        wildcard_match=self.wildcard_match,
-                        check_fn=self.check_fn,
-                        aws_conn_id=self.aws_conn_id,
-                        verify=self.verify,
-                        poke_interval=self.poke_interval,
-                    ),
-                    method_name="execute_complete",
-                )
+                self._defer()
+
+    def _defer(self) -> None:
+        """Check for a keys in s3 and defers using the triggerer."""
```

Comment:
this is not really what it's doing. It only does the "defers using the triggerer" part

maybe you meant to include the `if not self.poke(context=context):` in there ?

Reply:
Here `self.defer()` calls `self._defer()`. Poke happens in `self.defer()`